cmake/ecbuild_get_test_data.cmake

```diff
@@ -10,7 +10,7 @@
 
 # function for downloading test data
 
-function( _download_test_data _p_NAME _p_DIR_URL _p_DIRLOCAL _p_CHECK_FILE_EXISTS )
+function( _download_test_data _p_NAME _p_DIR_URL _p_DIRLOCAL _p_CHECK_FILE_EXISTS _p_INSECURE_CURL )
```

Comment:
This can be generalised to wget as well, the flag could be simply named `INSECURE`.

I think there is a case for also having a variable that controls that globally, on top of the case-by-case option (with the final value being true if either is set), say `ECBUILD_DOWNLOAD_INSECURE`.

Reply:
Done. I replaced `_p_INSECURE_CURL` with `_p_INSECURE`.
I have to admit that I do not know the purpose of the `_p_` prefix and applied it for consistency.

Reply:
Not sure either where the prefix comes from, but I would suspect it to be inspired by other functions that use [`cmake_parse_arguments`](https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html) with `_p_` as a prefix.